lib/chat_app_web/live/single_room_live.ex

```diff
@@ -7,23 +7,62 @@ defmodule ChatAppWeb.SingleRoomLive do
   @impl true
   def mount(%{"id" => room_id}, _session, socket) do
     topic = "room:" <> room_id
+
     if connected?(socket) do
-        Logger.info("Liveview's second mount() call")
-        LiveviewMonitor.monitor(self(), __MODULE__, %{id: socket.id, room_id: room_id})
-        PubSub.subscribe(ChatApp.PubSub, topic)
-        broadcasted_message = {:add_messages, ChatApp.Message.new("anon joined the chat.", "", room_id)}
-        PubSub.broadcast(ChatApp.PubSub, topic, broadcasted_message)
+      Logger.info("Liveview's second mount() call")
+      LiveviewMonitor.monitor(self(), __MODULE__, %{id: socket.id, room_id: room_id})
+      PubSub.subscribe(ChatApp.PubSub, topic)
+
+      message = %{
+        "content" => "anon joined the chat",
+        "sender" => "",
+        "room_id" => room_id,
+        "timestamp" => DateTime.utc_now(),
+        "is_deleted" => false,
+        "is_edited" => false,
+        "seq_number" => 0
+      }
+
+      IO.inspect(socket.assigns)
+
+      IO.inspect(message, label: "Broadcasted join message")
+
+      ChatApp.Contexts.Messages.insert_message(message)
+      PubSub.broadcast(ChatApp.PubSub, topic, :refresh_messages)
     else
-       Logger.info("Liveview's first mount() call")
+      Logger.info("Liveview's first mount() call")
     end
 
     room = ChatApp.Contexts.Rooms.get_room(room_id)
-    {:ok, assign(socket, room_id: room.id, room: room, topic: topic, messages: [], users: ChatApp.Contexts.Rooms.get_room_users(room.id).users, current_message: "")}
+
+    {:ok,
+     assign(socket,
+       room_id: room_id,
+       room: room,
+       topic: topic,
+       messages: ChatApp.Contexts.Rooms.get_room_messages(room_id),
+       users: ChatApp.Contexts.Rooms.get_room_users(room_id)
+     )}
   end
 
-  def unmount(%{id: _liveview_id, room_id: room_id}, _reason) do
-    broadcasted_message = {:add_messages, ChatApp.Message.new("anon left the chat.", "", room_id)}
-    PubSub.broadcast(ChatApp.PubSub, "room:" <> room_id, broadcasted_message)
+  def unmount(%{id: liveview_id, room_id: room_id}, _reason) do
+    message = %{
+      "content" => "anon left the chat",
+      "sender" => "",
+      "room_id" => room_id,
+      "timestamp" => DateTime.utc_now(),
+      "is_deleted" => false,
+      "is_edited" => false,
+      "seq_number" => 1
+    }
```

Comment:
what is `seq_number` ?
Also, this message is almost identical to the join message, maybe create a function that outputs this kind of messages.

Reply:
seq_number is not even used, that was just an idea that we had for ordering messages. I'll delete it.